vault/mount.go

```diff
@@ -350,19 +384,80 @@ func (c *Core) remount(src, dst string) error {
 	return nil
 }
 
+// tuneMount is used to set config on a mount point
+func (c *Core) tuneMount(path string, config MountConfig) error {
+	// Ensure we end the path in a slash
+	if !strings.HasSuffix(path, "/") {
+		path += "/"
+	}
+
+	// Prevent protected paths from being changed
+	for _, p := range protectedMounts {
+		if strings.HasPrefix(path, p) {
+			return fmt.Errorf("cannot tune '%s'", path)
+		}
+	}
+
+	// Verify exact match of the route
+	match := c.router.MatchingMount(path)
+	if match == "" || path != match {
+		return fmt.Errorf("no matching mount at '%s'", path)
+	}
+
+	// Find and modify mount
+	for _, ent := range c.mounts.Entries {
+		if ent.Path == path {
+			if config.MaxLeaseTTL != nil {
+				if *config.MaxLeaseTTL == 0 {
```

Comment:
Doesn't this logic always set the `et.Config.MaxLeaseTTL` to the `config.MaxLeaseTTL` value? Hard to see in the tree of ifs.